### PR TITLE
Flow extensions

### PR DESCRIPTION
**Pull request by ReneeVandervelde**
Adds some general purpose extensions for Flows.

### CODE REVIEW

File: coroutines/build.gradle.kts [hunk 1]
```diff
@@ -10,6 +10,14 @@ kotlin {
             }
         }
 
+        val jvmTest by getting {
+            dependencies {
+                implementation(Coroutines.test)
+                implementation(kotlin("test"))
```
Comment (AndrewReitz) [inline coroutines/build.gradle.kts:16]: The two kotlin deps would probably be better stuffed in the multiplatform-common

File: coroutines/src/commonMain/kotlin/com/inkapplications/coroutines/Flows.kt [hunk 2]
```diff
@@ -18,3 +19,59 @@ suspend inline fun <T> Flow<T>.safeCollect(crossinline action: suspend (T) -> Un
...
 }
+
+/**
+ * Collect a flow on a specific coroutine scope.
+ *
+ * This can be used to clean up some nesting with multiple launched
+ * coroutine flow collections.
+ * This uses [safeCollect] to ensure the scope is active when collecting.
+ */
+suspend fun <T> Flow<T>.collectOn(scope: CoroutineScope, action: suspend (T) -> Unit) {
```
Comment (AndrewReitz) [inline coroutines/src/commonMain/kotlin/com/inkapplications/coroutines/Flows.kt:30]: test missing for this one?

File: coroutines/build.gradle.kts (hunk 1 above)
Comment (ReneeVandervelde) [reply coroutines/build.gradle.kts:16]: Oh right, I already forgot you did that

File: coroutines/src/jvmTest/kotlin/com/inkapplications/coroutines/FlowsTest.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        }
+    }
+
+    @Test
+    fun collectOnTest() = runBlockingTest {
+        val fixedThread = Thread()
+        val scope = Executors.newSingleThreadExecutor { fixedThread }.asCoroutineDispatcher().let(::CoroutineScope)
+
+        flowOf(1).collectOn(scope) {
+            assertEquals(fixedThread, Thread.currentThread(), "Run on correct scope")
```
Comment (ReneeVandervelde) [inline coroutines/src/jvmTest/kotlin/com/inkapplications/coroutines/FlowsTest.kt:41]: @AndrewReitz This was the best way I could think of to test this.. it does work as a test.. but I kinda hate it?

Comment (AndrewReitz) [reply coroutines/src/jvmTest/kotlin/com/inkapplications/coroutines/FlowsTest.kt:41]: 🤮🤣

I donno as long as we know it works I guess it's good.

**Pull request merged**